### ISSUE TITLE
Certbot to update not only nginx.conf but config in sites-enabled

### ISSUE THREAD

**Issue by bertalanimre** [certbot/certbot#5186]
## My operating system is (include version):
CentOS 7

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
OS package manager (yum)

## I ran this command and it produced this output:
certbot --nginx

## Certbot's behavior differed from what I expected because:
Upgrade config files in /etc/nginx/sites-enabled folder as well

So my question is basically how can I tell certbot where to look for an nginx config to upgrade? Do I have to always copy all my server blocks into nginx.conf, upgrade with certbot, then copy the modifications back to my sites-enabled config file?

**Comment by bmw** [comment]
@ohemorange, can you help with this?

**Comment by ohemorange** [comment]
Hm, your expectations seem correct. The following information will help us debug your problem:

- The results of `tree /etc/nginx`
- The contents of any relevant config files, including `/etc/nginx/nginx.conf` and relevant files in `/etc/nginx/sites-enabled`
- The contents of relevant logs in `/var/log/letsencrypt/`

Please upload them to your favorite paste site (github gist works well), and feel free to redact any identifying information.

**Comment by fzipi** [comment]
Having the same problem. Ubuntu 16.04.3 LTS

https://gist.github.com/fzipi/248f8b076753d36663f63f0990a8d5b3

**Comment by bertalanimre** [comment]
Excuse me for being AFK, had some accident and still recovering. Therefore I'm unable to provide any information right now. However, I can describe that you ran replicate your issue if you leave the nginx.conf file standard as it is, only adding another line to include the `sites-enabled` folder and create a standard laravel nginx configuration inside sites-enabled. That is what we use.

Normally we also have an ssl folder inside /etc/nginx to store the ssl certificates, but since certbot is here to exclude this, I'm just mentioning this. Sorry again for nor replying.Gonna recover myself and my car first. :(

**Comment by ohemorange** [comment]
So sorry to hear about that! @fzipi, since it sounds like you might have a similar problem, could you provide the contents of `/etc/nginx/nginx.conf` as well, and the contents of `/var/log/letsencrypt/letsencrypt.log`?

**Comment by fzipi** [comment]
Well.. Could it be that order of directives in the included file is important for parsing?

Moving `server_name` to the beggining makes it work!
```
*** sites-available/owaspsummit-test.org.conf.notworking	2017-11-02 01:57:17.927863168 +0000
--- sites-available/owaspsummit-test.org.conf	2017-11-02 01:57:35.087880673 +0000
***************
*** 14,19 ****
--- 14,21 ----
  # Default server configuration
  #
  server {
+ 	server_name owaspsummit-test.org;
+ 
  	listen 80;
  	listen [::]:80;
  
***************
*** 37,44 ****
  
  	# Add index.php to the list if you are using PHP
  	index index.html;
- 	server_name owaspsummit-test.org;
- 
  
  	location / {
  		# First attempt to serve request as file, then
--- 39,44 ----
```

I'm assembling a full not working example now.

**Comment by fzipi** [comment]
[Here](https://gist.github.com/fzipi/2fd64dd58f4c1768763279599098199c) is the base nginx config.

**Comment by bertalanimre** [comment]
Mine are:

`/etc/nginx/sites-enabled/hu.mycompany.mywebapp.conf`

```
map $http_user_agent $limit_bots {
     default 0;
     ~*(google|bing|yandex|msnbot) 1;
     ~*(AltaVista|Googlebot|Slurp|BlackWidow|Bot|ChinaClaw|Custo|DISCo|Download|Demon|eCatch|EirGrabber|EmailSiphon|EmailWolf|SuperHTTP|Surfbot|WebWhacker) 1;
     ~*(Express|WebPictures|ExtractorPro|EyeNetIE|FlashGet|GetRight|GetWeb!|Go!Zilla|Go-Ahead-Got-It|GrabNet|Grafula|HMView|Go!Zilla|Go-Ahead-Got-It) 1;
     ~*(rafula|HMView|HTTrack|Stripper|Sucker|Indy|InterGET|Ninja|JetCar|Spider|larbin|LeechFTP|Downloader|tool|Navroad|NearSite|NetAnts|tAkeOut|WWWOFFLE) 1;
     ~*(GrabNet|NetSpider|Vampire|NetZIP|Octopus|Offline|PageGrabber|Foto|pavuk|pcBrowser|RealDownload|ReGet|SiteSnagger|SmartDownload|SuperBot|WebSpider) 1;
     ~*(Teleport|VoidEYE|Collector|WebAuto|WebCopier|WebFetch|WebGo|WebLeacher|WebReaper|WebSauger|eXtractor|Quester|WebStripper|WebZIP|Wget|Widow|Zeus) 1;
     ~*(Twengabot|htmlparser|libwww|Python|perl|urllib|scan|Curl|email|PycURL|Pyth|PyQ|WebCollector|WebCopy|webcraw) 1;
 } 

server	{
 	listen	80;
 
 	server_name	mywebapp.mycompany.hu;

	return		301 https://mywebapp.mycompany.hu$request_uri;
}
 
server	{
 	listen			443 ssl;
 	server_name		mywebapp.mycompany.hu;
 
 	ssl_certificate 	/etc/nginx/ssl/mycompany.hu.combined.crt;
 	ssl_certificate_key	/etc/nginx/ssl/mycompany.hu.key;

	ssl_dhparam /etc/nginx/ssl/dhparams.pem;
	
	ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
	
	ssl_prefer_server_ciphers on;
	
	ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:ECDHE-RSA-AES128-GCM-SHA256:AES256+EECDH:DHE-RSA-AES128-GCM-SHA256:AES256+EDH:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA:ECDHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES256-GCM-SHA384:AES128-GCM-SHA256:AES256-SHA256:AES128-SHA256:AES256-SHA:AES128-SHA:DES-CBC3-SHA:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!MD5:!PSK:!RC4";
	
	add_header Strict-Transport-Security "max-age=63072000; includeSubdomains; preload";
 
 	root 			/var/www/html/hu.mycompany.mywebapp/public;
 
 	error_log 		/var/log/nginx/hu.mycompany.mywebapp/error.log;
 	access_log		/var/log/nginx/hu.mycompany.mywebapp/access.log;
 
 	index			index.php;

#	auth_basic "Restricted";
#	auth_basic_user_file /etc/nginx/.htpasswd.hu.mycompany.mywebapp;

	gzip  on;
	gzip_vary on;
	gzip_min_length 10240;
	gzip_proxied expired no-cache no-store private auth;
	gzip_types text/plain text/css text/xml text/javascript application/x-javascript application/xml;
	gzip_disable "MSIE [1-6]\.";

	location / {
		try_files $uri $uri/ /index.php?$query_string;
		if ($limit_bots = 1) {
			return 403;
		}
	}
	
	location ~* \.(jpg|jpeg|gif|png|css|js|ico|xml)$ {
		access_log        off;
		log_not_found     off;
		expires           1d;
	}

	location ~ /index.php(/|$) {
		try_files $uri =404;
		fastcgi_pass   unix:/run/php-fpm/mywebapp.sock;
		include        fastcgi_params;
		fastcgi_param  SCRIPT_FILENAME  $document_root$fastcgi_script_name;
		fastcgi_split_path_info ^(.+\.php)(/.+)$;
		fastcgi_index  index.php;
		fastcgi_buffers 16 16k;
		fastcgi_buffer_size 32k;
		fastcgi_read_timeout 300;
	}
}
```

`/etc/nginx/nginx.conf`

```
user web;
worker_processes  4;

error_log	/var/log/nginx/error.log warn;
pid		/var/run/nginx.pid;


events {
		worker_connections  1024;
}


http {

		include       /etc/nginx/mime.types;
		default_type  application/octet-stream;

		log_format  main  '$remote_addr - $remote_user [$time_local] "$request" '
		                  '$status $body_bytes_sent "$http_referer" '
		                  '"$http_user_agent" "$http_x_forwarded_for"';
		
		access_log  /var/log/nginx/access.log  main;
		
		client_max_body_size 20M;
	
		fastcgi_buffers 8 16k;
		fastcgi_buffer_size 32k;
	
		sendfile        on;
		#tcp_nopush     on;
		
		keepalive_timeout  65;
		
		#include /etc/nginx/conf.d/*.conf;
		include /etc/nginx/sites-enabled/*.conf;
		
		gzip on;
		gzip_min_length 1100;
		gzip_buffers 4 32k;
		gzip_types text/plain application/x-javascript text/xml text/css;
		gzip_vary on; 
	
		server 
		{
			listen	80	default_server;
			server_name _;
			return      https://mycompany.hu;
		}
}
```

**Comment by ohemorange** [comment]
Hi all. Are these issues still persisting? If so, could you please upload the rest of the debugging information:

- The results of `tree /etc/nginx`
- The contents of relevant logs inside `/var/log/letsencrypt/`

**Comment by bertalanimre** [comment]
I'll close this ticket cause I think this is not relevant nowdays. I've just installed a webserver and applied Certbot with ease without thinking about it. The only think I've customized my setup is that I've created soft link for the certificate and key files from the letsencrypts defaults to my /etc/nginx/ssl folder. Other than that, it works flawless. I know I didn't give 2 cr*p about my own topic and I feel sorry about it. I just had too many other stuff to do but a HUGE Thank You for Everyone who helped in it.
